### PR TITLE
Apply event sourcing to event sub process processor

### LINKED ISSUE #6196: Apply event-sourcing to the BPMN event subprocess processor
**Description**

* apply the event-sourcing concept to the BPMN event subprocess processor

Open questions:
* Can we align the sequence of records of an event sub process with regular processes? (activate the process vs. the start event first)
* How do we store the variables when a (message) event sub process is triggered? (the message is correlated before the event sub process is activated)

Reference:
* [ZEP: Event Subprocess](https://github.com/zeebe-io/enhancements/blob/master/ZEP004-wf-stream-processing.md#event-sub-process)

### PR DESCRIPTION
**Pull request by korthout**
## Description

Migrates the event sub process processor.

When an event triggers, it now activates the event sub process by writing the `activating` and `activated` event records directly, instead of writing the `activate` command. This is necessary to correctly copy the variables of the triggered event into the event sub process' scope. To continue the execution, the start event is activated by writing the `activate` command for it.

Since the activation of the event sub process is not processed by the `EventSubProcessProcessor`, the input mappings of the event sub process are evaluated on the processing of the start event `activate` command. 

> Note, it would not have been possible to evaluate this as part of the event trigger processing, because input mappings can cause incidents and incidents can only exist on elements. For example, it would not be possible to resolve an incident for a failure during Timer:Trigger processing.

Incidents due to the evaluation of the input mappings of the event sub process are thus created on the start event, to allow the start event `activate` command to be retried when resolving the incident.

## Related issues

closes #6196

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: engine/src/main/java/io/zeebe/engine/processing/common/EventTriggerBehavior.java [hunk 1]
```diff
@@ -342,4 +254,25 @@ public void activateTriggeredEvent(
     commandWriter.appendFollowUpCommand(
         eventInstanceKey, ProcessInstanceIntent.COMPLETE_ELEMENT, eventRecord);
   }
+
+  private void activateEventSubProcess(
+      final ExecutableFlowElement triggeredEvent, final ExecutableFlowElement flowScope) {
+    // first activate the event sub process
```
Comment (Zelldon) [inline engine/src/main/java/io/zeebe/engine/processing/common/EventTriggerBehavior.java:260]: Lets improve the comment such that we understand also later why we not write the command here and immediately go to ACTIVATED and add that we write the ACTIVATE command for the start event.

Comment (Zelldon) [reply engine/src/main/java/io/zeebe/engine/processing/common/EventTriggerBehavior.java:260]: I can also do it, its just a reminder.

File: engine/src/main/java/io/zeebe/engine/processing/bpmn/container/EventSubProcessProcessor.java [hunk 2]
```diff
@@ -43,91 +42,50 @@ public EventSubProcessProcessor(final BpmnBehaviors bpmnBehaviors) {
...
+  public Either<Failure, ?> onChildActivating(
       final ExecutableFlowElementContainer element,
       final BpmnElementContext flowScopeContext,
-      final BpmnElementContext childContext) {}
+      final BpmnElementContext childContext) {
+    if (childContext.getBpmnElementType() != BpmnElementType.START_EVENT) {
+      return Either.right(null);
+    }
+
+    return variableMappingBehavior.applyInputMappings(flowScopeContext, element);
```
Comment (Zelldon) [inline engine/src/main/java/io/zeebe/engine/processing/bpmn/container/EventSubProcessProcessor.java:87]: @korthout do you think we need a comment here, which says why we actually do these input mapping here and have no activate command?

**Review by ChrisKujawa — APPROVED** [review]
Awesome with that we are done :checkered_flag: :100:

File: engine/src/main/java/io/zeebe/engine/processing/common/EventTriggerBehavior.java (hunk 1 above)
Comment (korthout) [reply engine/src/main/java/io/zeebe/engine/processing/common/EventTriggerBehavior.java:260]: Sure, that makes total sense

**Comment by Zelldon** [comment]
bors r+ 

I will create a separate PR

**Pull request merged**